// We had a bug where this gave the incorrect error: "Expected a number but found 'a' at definition 'Foo'."
data bool = false | true

(Foo false a) = 0
(Foo true 0) = 0
(Foo true n) = n

Main = (Foo true 3)